packages/flutter/lib/src/widgets/scroll_controller.dart

```diff
@@ -12,7 +12,34 @@ import 'scroll_physics.dart';
 import 'scroll_position.dart';
 import 'scroll_position_with_single_context.dart';
 
+/// Controls a scrollable widget.
+///
+/// Scroll controller are typically stored as member variables in [State]
+/// objects and are reused in each [State.build]. A single scroll controller can
+/// be used to control multiple scrollable widgets, but some operations, such
+/// as reading the scroll [offset], require the controller to be used with a
+/// single scrollable widget.
+///
+/// A scroll controller creates a [ScrollPosition] to manage the state specific
+/// to an individual [Scrollable] widget. To use a custom [ScrollPosition],
+/// subclass [ScrollController] and override [createScrollPosition].
+///
+/// Typically used with [ListView], [GridView], [CustomScrollView].
+///
+/// See also:
+///
+///  * [ListView], [GridView], [CustomScrollView], which can be controlled by a
+///    [ScrollController].
+///  * [Scrollable], which is the lower-level widget that creates and associates
```

Comment:
I'm not sure it makes sense to mention Scrollable here. ScrollView explains the relationship between FooView and Scrollable, should the reader get that far.